apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,34 +1,19 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { NgFor, NgIf, NgTemplateOutlet } from '@angular/common';
+import { Component, ContentChild, Input, TemplateRef } from '@angular/core';
 import { ListItemComponent } from '../list-item/list-item.component';
 
 @Component({
   selector: 'app-card',
   templateUrl: './card.component.html',
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  imports: [NgIf, NgFor, ListItemComponent, NgTemplateOutlet],
 })
 export class CardComponent {
   @Input() list: any[] | null = null;
-  @Input() type!: CardType;
   @Input() customClass = '';
+  @ContentChild(TemplateRef) nameTemplate: TemplateRef<{
+    item: any;
+  }> | null = null;
```

Comment:
That's very dangerous, what will happen if you had another template inside your parent. 

Use a directive to call the right template, and use the `read` property to transform it to TemplateRef

Reply:
i will try that out

Reply:
i have made use of read property to enable the component to access multiple parent templates, thanks a lot for the suggestion